keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/MemoryCloudDriverCache.kt

```diff
@@ -138,6 +139,15 @@ class MemoryCloudDriverCache(
     notFoundMessage: String,
     loader: suspend CoroutineScope.() -> T?
   ): T = get(key) {
-    runBlocking(block = loader)
+    runCatching {
+      runBlocking(block = loader)
+    }
+      .getOrElse { ex ->
+        if (ex is HttpException && ex.code() == 404) {
+          null
+        } else {
+          throw CacheLoadingException("Error loading cache for $key", ex)
+        }
+      }
   } ?: throw ResourceNotFound(notFoundMessage)
```

Comment:
Should we _ever_ throw `ResourceNotFound` exceptions from the cache, since none of the objects it handles are actually exposed as first-class objects in keel's APIs? Maybe we need a new type for this case too (`ReferencedResourceNotFound`?).

Reply:
Yeah, that would make sense